### DISCUSSION TITLE
Can't use "DEFINE COLUMN" since I connect using Direct Query, can't connect using import mode

### DISCUSSION THREAD

**Discussion by simon-benoit** [DaxStudio/DaxStudio#714] [Q&A]
AFAIK, DEFINE COLUMN is not available when using Direct Query (as per https://docs.microsoft.com/en-us/analysis-services/tabular-models/understanding-dax-in-tabular-models-ssas-tabular?view=asallproducts-allversions ) 

Where can I connect to my tabular model using import mode?

Connect / Advanced options / Direct Query Mode only offers "Default", "In memory" or "DirectQuery", no "import".

Whatever mode I choose, I always get an error when trying to use DEFINE COLUMN : "cannot create column _____ in table  _____ since query scope calculated column is not supported"

**Comment by dgosbell** [comment]
> Where can I connect to my tabular model using import mode?

You cannot switch the mode of a model on the fly. This is really a legacy feature now, they've even removed these settings from the UI in SQL Management Studio so I might do this also this is the explanation for what these settings meant:
https://docs.microsoft.com/en-us/previous-versions/sql/sql-server-2012/hh230898(v=sql.110)?redirectedfrom=MSDN#directquery-properties

And even if it was not a legacy feature changing this setting would not help since it basically just changed which partitions were used to respond to a query, but it still enforced all the same DAX limitations that the full Direct Query mode since models built in this way had to support both modes. You also had to have the in-memory/import partitions built and refreshed ahead of time in order to access them.

So if your model has been built in Direct Query mode you will not be able to use DEFINE COLUMN and there is nothing you can do from the DAX Studio side of things about this. You would need to look at changing storage mode of the model which may not be a simple thing depending on why DirectQuery was chosen in the first place.
